src/vanilla/utils/atomWithDefault.ts

```diff
@@ -8,19 +8,20 @@ type Read<Value, Args extends unknown[], Result> = WritableAtom<
   Result
 >['read']
 
-export function atomWithDefault<Value>(
-  getDefault: Read<Value, [SetStateAction<Value> | typeof RESET], void>,
-): WritableAtom<Value, [SetStateAction<Value> | typeof RESET], void> {
+export function atomWithDefault<Value, KType = Awaited<Value>>(
```

Comment:
This doesn't feel correct and overcomplicated.

I might be wrong. If there's any other issue, please share typescript playground reproduction.